torchci/pages/api/drci/drci.ts

```diff
@@ -92,30 +192,66 @@ export function constructResultsComment(
             output += somePending;
         }
         output += `\nAs of commit ${sha}:`;
-        output += '\n<details open><summary>The following jobs have failed:</summary><p>\n\n';
-        const failedJobsSorted = failedJobs.sort((a, b) => a.name.localeCompare(b.name))
-        for (const job of failedJobsSorted) {
-            output += `* [${job.name}](${job.html_url})\n`;
-        }
-        output += "</p></details>"
+        output += constructResultsJobsSections(
+          "The following jobs have failed",
+          failedJobs
+        );
     }
+    output += constructResultsJobsSections(
+      "The following jobs failed but were likely due to flakiness present on master",
+      flakyJobs
+    );
+    output += constructResultsJobsSections(
```

Comment:
I think another way to see if the failure is due to broken trunk is to use the same mechanism we have to send oncall alert, i.e. the Rockset query behind https://github.com/pytorch/test-infra/issues/1254.  IMO, this is more reliable because the alarm ensures that two or more similar failures occur consecutively in master instead of just the merge base commit. The code also doesn't need to call `addMergeBaseCommits` to find out the merge base commit and query all the associated jobs.  On the other hand, the current approach is probably simpler to implement.  Both have their merits, so we can keep this implement till there is a need to optimize it or if there are false positives for broken trunk

Reply:
thats a good idea, it would also get around not being able to get periodic workflow errors since theyre usually not on the base commit, one problem i see happening is that you might have to query pretty far back since I saw a pr that got merged this week that was failing due to something that broke maybe 2 weeks ago, but perhaps the query could be changed to be able to get stuff that far back

Reply:
Yeah, that's exactly the crux of it.  Do we have a rule to prevent merging if the base commit is older than X days?  If yes, we would just need to query that far back.  I remember someone mentions 3 days (and I wonder if this applies to ghstack)

Reply:
i just took a look at trymerge, the 3 day threshold is for last push date, so we don't actually care about the age of the base commit

Reply:
Oh, I see. TIL.  In hindsight, may be we should have that check.  Merging a PR based on month-old base commit sounds like asking for trouble.

On the other hand, let's see if we can dig out the p90 or P95 of the difference in days (or hours) between the commit and the base commit.  That's probably a reasonable number to use IMO